lib/engine/game/g_18_ts/game.rb

```diff
@@ -0,0 +1,1166 @@
+              player.spend(company.value, @bank)
+              player.companies << company
+              company.owner = player
+            elsif idx == size
+              prr = corporation_by_id('PRR')
+              price = par_prices(prr).first
+              @stock_market.set_par(prr, price)
+              share = prr.ipo_shares.first
+              @share_pool.buy_shares(player, share.to_bundle, exchange: nil, swap: nil, allow_president_change: true)
+              after_par(prr)
+            else
+              prr = corporation_by_id('PRR')
+              share = prr.ipo_shares.find { |s| !s.owner&.player? }
+              @share_pool.buy_shares(player, share.to_bundle, exchange: nil, swap: nil, allow_president_change: true)
+            end
+          end
+          @highest_layer = 1 if unbought_companies.empty?
+
+          # pull out minor trains from depot
+          @minor_trains = []
+          corporations.each do |minor|
+            next unless (name = REQUIRED_TRAIN[minor.name])
+
+            req_train = @depot.upcoming.find { |t| t.name == name }
+            raise GameError, "Unable to find train #{name} for minor #{minor.name}" unless req_train
+
+            req_train.buyable = false
+            req_train.reserved = true
+            @minor_trains << req_train
+            @depot.remove_train(req_train)
+          end
+
+          # pre-allocate dummy trains used for tile 200
+          @pass_thru = {}
+          DUMMY_TRAINS.each { |train| @pass_thru[train[:name]] = Train.new(**train, requires_token: false, index: 999) }
+        end
+
+        # cache all stock prices
+        def share_prices
+          stock_market.market.first
+        end
+
+        def active_players
+          players_ = @round.active_entities.map(&:player).compact
+
+          players_.empty? ? acting_when_empty : players_
+        end
+
+        def acting_when_empty
+          if (active_entity = @round && @round.active_entities[0])
+            [acting_for_entity(active_entity)]
+          else
+            @players
+          end
+        end
+
+        # for receivership:
+        # find first player from PD not a director
+        # o.w. PD
+        def acting_for_entity(entity)
+          return entity if entity&.player?
+          return entity.owner if entity&.owner&.player?
+
+          acting = @players.find { |p| !director?(p) }
+          acting || @players.first
+        end
+
+        def director?(player)
+          @corporations.any? { |c| c.owner == player }
+        end
+
+        def tile_valid_for_phase?(tile, hex: nil, phase_color_cache: nil)
+          phase_color_cache ||= @phase.tiles
+
+          # 119 upgrades from yellow in phase 3
+          return false if tile.name == '119' && !phase_color_cache.include?(:brown)
+
+          # 166 upgrades from green in phase 4
+          return false if tile.name == '166' && !phase_color_cache.include?(:gray)
```

Comment:
Are these needed? 119 is a green tile with brown stripes, so should automatically be available when green tiles are available. Likewise 166 is a brown tile with grey stripes, so should become available at the same time as brown tiles.